datajoint/autopopulate.py

```diff
@@ -89,6 +89,7 @@ def populate(self, *restrictions, suppress_errors=False, reserve_jobs=False, ord
             keys = list(keys)
             random.shuffle(keys)
 
+        logger.info('Found %d keys to populate' % len(todo))
```

Comment:
`len(todo)` is an extra query, which may take some time.  Would you extract the length from `keys` instead?